From ca10db58cce7f4f594b2d9fe9be008e6ade52f02 Mon Sep 17 00:00:00 2001
From: "qiang.cheng" <qiang.cheng@amlogic.com>
Date: Tue, 18 Jul 2023 17:08:34 +0800
Subject: [PATCH 07/15] uart: fix uart too many interrupt [1/1]

PD#SWPL-131844

Problem:
BT uart too many interrupt, lead to BT work abnormal

Solution:
use interrupt bottom half

Verify:
sm1_debian t7c_debian

Signed-off-by: qiang.cheng <qiang.cheng@amlogic.com>
Change-Id: I7427f65a3d29afe11c1d2831b74d0b4280fc9485
---
 arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts           | 1 +
 arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts          | 1 +
 arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts           | 1 +
 .../boot/dts/amlogic/t7_a311d2_an400_drm_hdmitx_only_debian.dts  | 1 +
 arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts          | 1 +
 .../amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts    | 1 +
 6 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
index 349559f8b..ddfdd3827 100644
--- a/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/g12b_a311d_w400_debian.dts
@@ -1564,6 +1564,7 @@
 
 &uart_A {
 	status = "okay";
+	uart_for_bt = <1>;
 };
 
 &pcie_A {
diff --git a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
index 86771c783..fe6bfddf6 100644
--- a/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/sm1_s905d3_ac200_debian.dts
@@ -1870,6 +1870,7 @@
 
 &uart_A {
 	status = "okay";
+	uart_for_bt = <1>;
 };
 
 /*if you want to use vdin just modify status to "okay"*/
diff --git a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
index 65d534a46..8d5d5c279 100644
--- a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_debian.dts
@@ -1882,6 +1882,7 @@
 &uart_C {
 	status = "okay";
 	uart-has-rtscts;
+	uart_for_bt = <1>;
 };
 
 &pwmao_ab {
diff --git a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_drm_hdmitx_only_debian.dts b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_drm_hdmitx_only_debian.dts
index 010e279c3..2aaa6c38d 100644
--- a/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_drm_hdmitx_only_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7_a311d2_an400_drm_hdmitx_only_debian.dts
@@ -1777,6 +1777,7 @@
 &uart_C {
 	status = "okay";
 	uart-has-rtscts;
+	uart_for_bt = <1>;
 };
 
 &pwmao_ab {
diff --git a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
index 221f35987..b6b41e756 100644
--- a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_debian.dts
@@ -2042,6 +2042,7 @@
 &uart_C {
 	status = "okay";
 	uart-has-rtscts;
+	uart_for_bt = <1>;
 };
 
 &pwmao_ab {
diff --git a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
index cc23d0439..3a9da4758 100644
--- a/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
+++ b/arch/arm64/boot/dts/amlogic/t7c_a311d2_an400_linux_drm_hdmitx_only_debian.dts
@@ -1983,6 +1983,7 @@
 &uart_C {
 	status = "okay";
 	uart-has-rtscts;
+	uart_for_bt = <1>;
 };
 
 &pwmao_ab {
-- 
2.25.1

